rust/core-lib/src/editor.rs

```diff
@@ -818,3 +832,12 @@ fn n_spaces(n: usize) -> &'static str {
     assert!(n <= spaces.len());
     &spaces[..n]
 }
+
+/// Counts the number of lines in the string, not including any trailing newline.
+fn count_lines(s: &str) -> usize {
+    let mut newlines = count_newlines(s);
+    if s.as_bytes().last() == Some(&0xa) {
```

Comment:
Do we want to/should we bother with CR/LF/CRLF?

Reply:
This handles LF and CRLF. I don't believe we're treating CR on its own as a newline? At least we aren't doing it elsewhere in the codebase that I can determine.

Reply:
Short answer, yes. Longer answer, I'm hoping that CR-only files will be usable with word-wrapping on, as the line break iterator will always indicate them as hard breaks. That's not quite the same, because it won't count in line numbering, but I think is good enough considering the likelihood of the files in the wild.